spec/std/spec/junit_formatter_spec.cr

```diff
@@ -124,14 +154,19 @@ private def build_report
   output = String::Builder.new
   formatter = Spec::JUnitFormatter.new(output)
   yield formatter
-  formatter.finish
-  output.to_s
+  formatter.finish(Time::Span.zero, false)
+  output.to_s.chomp
 end
 
-private def exception_with_backtrace(msg)
-  begin
-    raise Exception.new(msg)
-  rescue e
-    e
+private def build_report_with_no_timestamp
+  output = build_report do |formatter|
+    yield formatter
   end
+  output.gsub(/\s*timestamp="(.+?)"/, "")
```

Comment:
I'd rather add a way to inject `@started_at` instead of tampering with the output.

The current workaround completely avoids testing the correctness of the `timestamp` attribute.

Reply:
I concur, yet I'd do it as a part of a rewrite of the `Spec::JUnitFormatter` and its specs, to use `XML` library and more robust testing (not based on exact source code like it is atm)

Reply:
Okay, so you're just adding a few features in this PR and plan to refactor the entire code later?
Why not refactor first?

Also please be aware that the JUnit formatter does not use `XML` *on purpose* to avoid requireing `libxml` as a dependency for every spec run (see #6004).

Reply:
> Why not refactor first?
> 
> Also please be aware that the JUnit formatter does not use `XML` _on purpose_ to avoid requireing `libxml` as a dependency for every spec run (see #6004).

Exactly because of that. I'd like to re-open discussion about it. Crafting XML "by hand" comes hardly as a proper solution here.

Reply:
Well, re-adding libxml is not an acceptable solution either. This is not a proper solution, but it does work. Crafting a basic format by hand is fine.

Reply:
@straight-shoota How about adding specific test just for `timestamp` attribute? This way we could have more-or-less readable specs and tested attribute in the same time...

Reply:
How's that better/easier to implement than mocking the timestamp property?

Reply:
Mocking timestamp throughout the tests would necessitate adding this to every tests using exact matchers. Also, see my newest commit.